### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by bashtage**
Fixes all future warnings
Adds a Python 3.7 test run
Fixes errors that appear on the Python 3.7 test run
Adds explicit OpenBlas testing.

Edit: Refactored the commit history to split concerns more cleanly

### CODE REVIEW

**Comment by bashtage** [comment]
closes #4779 
closes #4821

File: statsmodels/genmod/cov_struct.py [hunk 1]
```diff
@@ -396,8 +396,8 @@ def initialize(self, model):
             # This is used to construct the working correlation
             # matrix.
             ilabel = np.zeros((ngrp, ngrp), dtype=np.int32)
-            ilabel[[ix1, ix2]] = ncm + 1
-            ilabel[[ix2, ix1]] = ncm + 1
+            ilabel[(ix1, ix2)] = ncm + 1
```
Comment (serhiy-storchaka) [inline statsmodels/genmod/cov_struct.py]: `ilabel[ix1, ix2]` works as well.

File: statsmodels/iolib/summary2.py [hunk 2]
```diff
@@ -223,7 +223,7 @@ def as_latex(self):
 
         if self._merge_latex:
             # create single tabular object for summary_col
-            tab = re.sub(to_replace,'\midrule\n\midrule\n', tab)
+            tab = re.sub(to_replace,'\\\midrule\n\\\midrule\n', tab)
```
Comment (serhiy-storchaka) [inline statsmodels/iolib/summary2.py]: Shouldn't it be `'\\\\midrule\n\\\\midrule\n'` or `r'\midrule\n\midrule\n'`?

File: statsmodels/tsa/statespace/representation.py [hunk 3]
```diff
@@ -381,7 +381,7 @@ def __getitem__(self, key):
 
             # See note on time-varying arrays, below
             if matrix.shape[-1] == 1:
-                return matrix[[slice(None)]*(matrix.ndim-1) + [0]]
+                return matrix[tuple([slice(None)]*(matrix.ndim-1) + [0])]
```
Comment (serhiy-storchaka) [inline statsmodels/tsa/statespace/representation.py]: Or ` (slice(None),)*(matrix.ndim-1) + (0,) `.

File: statsmodels/iolib/summary2.py (hunk 2 above)
Comment (bashtage) [reply statsmodels/iolib/summary2.py]: You would think so but Python 3.7 wouldn't allow `\\midrule` which should be the correct statement.

Comment (serhiy-storchaka) [reply statsmodels/iolib/summary2.py]: Sorry, I meant `r'\\midrule\n\\midrule\n'`. You need to use only 2 backslashes in a raw string literal.

The `\n` escape sequences is interpreted in the replacement string, so both `'\n'` and `r'\n'` mean the same.

File: statsmodels/imputation/mice.py [hunk 4]
```diff
@@ -1078,8 +1078,8 @@ def impute_pmm(self, vname):
 
         # Unwind the indices
         jj = np.arange(dxi.shape[0])
-        ix = dxi[[jj, ir]]
-        iz = ixm[[jj, ix]]
+        ix = dxi[(jj, ir)]
```
Comment (serhiy-storchaka) [inline statsmodels/imputation/mice.py]: Same as above, can be `dxi[jj, ir]`.

**Comment by josef-pkt** [comment]
This is or was all green, so merging (with a recovering from vacation delay)

@bashtage  Thank you

**Pull request merged**